frappe/public/js/form_builder/components/Field.vue

```diff
@@ -53,6 +55,65 @@ function duplicate_field() {
 	props.column.fields.splice(index + 1, 0, duplicate_field);
 	store.form.selected_field = duplicate_field.df;
 }
+function make_dialog (frm) {
+	frm.dialog = new frappe.ui.Dialog({
+		title: __("Select Filters"),
+		fields: [
+			{
+				fieldtype: "HTML",
+				fieldname: "filter_area",
+			},
+		],
+		primary_action: () => {
+			let filters = frm.filter_group.get_filters().map(filter =>{
+				// last element is a boolean which hides the filter hence not required to store in meta
+				filter.pop()
+				filter[0] = props.field.df.fieldname
+				return filter
+			});
+
+			props.field.df.link_filters = JSON.stringify(filters);
+			frm.dialog.hide();
+		},
+		primary_action_label: __("Apply")
+	});
+};
+
+function make_filter_area (frm,doctype) {
+	frm.filter_group = new frappe.ui.FilterGroup({
+		parent: frm.dialog.get_field("filter_area").$wrapper,
+		doctype: doctype,
+		on_change: () => {},
+	});
+}
+
+function add_existing_filter(df){
+	if (df.link_filters){
+		let filters = JSON.parse(df.link_filters);
+		filters.map(filter => {
+			filter[0] = frappe.unscrub(filter[0])
```

Comment:
Why is this needed? Are you trying to guess the fieldname here?

Reply:
Here, we are trying to find if that field has existing filters applied on it.
If filters are there

then we will first JSON.parse it

then change the first element in each filter to "unscrub", because that is the required format of Filter Group UI Component.